src/pages/Units/NewUnitsPage.jsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import firebase from 'firebase/app';
+import PageTitle from '../../atoms/PageTitle';
+import 'firebase/firestore';
+import CodexFilter from '../../molecules/CodexFilter';
+import SubmitButton from '../../atoms/SubmitButton';
+import TextInputRow from '../../atoms/TextInputRow';
+import SelectArray from '../../molecules/SelectArray';
+import store from '../../Redux/store';
+import codexFilter from '../../utilities/codexFilter';
+import NumberInputRow from '../../atoms/NumberInputRow';
+
+const NewUnitsPage = () => {
+  const [newUnit, setNewUnit] = React.useState({
+    Codex: store.getState().codex, Name: '', Cost: 0, Abilities: 'None', Gear: [], userID: store.getState().user.uid,
+  });
+
+  function handleNameInput(input) { setNewUnit({ ...newUnit, Name: input }); }
+  function handleCostInput(input) { setNewUnit({ ...newUnit, Cost: +input }); }
+  function handleAbilitiesInput(input) { setNewUnit({ ...newUnit, Abilities: input }); }
+  function handleGearRemove(input) {
+    const NewGear = newUnit.Gear.filter((item) => item !== input);
+    setNewUnit({ ...newUnit, Gear: NewGear });
+  }
+  function handleGearAdd(input) {
+    const NewGear = newUnit.Gear;
+    NewGear.push(input);
+    setNewUnit({ ...newUnit, Gear: NewGear });
+  }
+  function handleNewUnitSubmission(event) {
+    event.preventDefault();
+    firebase.firestore().collection('units').add(newUnit);
+    window.location.hash = '/units/view';
+  }
+
+  // eslint-disable-next-line
+    React.useEffect(()=>setNewUnit({...newUnit,Codex:store.getState().codex}),[store.getState().codex]);
+
+  return (
+    <div data-test="newUnitsPage" className="container-fluid p-padding text-center">
+      <PageTitle Title="New Units Page" />
+      <form data-test="submitButton" onSubmit={(e) => handleNewUnitSubmission(e)}>
+        <CodexFilter />
+        <TextInputRow type="text" left="New Unit Name:" startValue={newUnit.Name} onInputChange={handleNameInput} />
+        <NumberInputRow left="New Unit Cost:" startValue={newUnit.Cost} onInputChange={handleCostInput} />
+        <TextInputRow type="text" left="New Unit Abilities:" startValue={newUnit.Abilities} onInputChange={handleAbilitiesInput} />
```

Comment:
Line breaks for your HTML props are highly recommended. Ideal HTML line breaks should look like this.

```
<TextInputRow 
  type="text" 
  left="New Unit Abilities:" 
  startValue={newUnit.Abilities} 
  onInputChange={handleAbilitiesInput} 
/>
```